### PR TITLE
Removed legacy auxilliary thread MessageLogger feature

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

The MessageLogger had a feature to run a separate thread to process the logging messages. This feature has never actually been used outside of tests. Removing it makes maintenance and future extensions to the MessageLogger easier.

#### PR validation:

Related Geometry and Framework unit tests all pass.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Framework/bin/cmsRun.cpp [hunk 1]
```diff
@@ -135,23 +134,10 @@ int main(int argc, char* argv[]) {
       //    be run below and can lead to error messages which should be sent via
       //    the message logger)
       context = "Initializing either multi-threaded or single-threaded message logger";
-      bool multiThreadML = false;
```
Comment (wddgit) [inline FWCore/Framework/bin/cmsRun.cpp:138]: On previous line, does the context string need to be edited to remove multi-threaded?

Comment (Dr15Jones) [reply FWCore/Framework/bin/cmsRun.cpp:138]: Good point

**Comment by makortel** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Comment by mrodozov** [comment]
somehow this PR got ~70 unit test and ~800 relvals broken with the following:

```
Thread 1 (Thread 0x100003530000 (LWP 49116)):
#0  0x0000100002bce4e8 in waitpid () from /lib64/libc.so.6
#1  0x0000100002b3fd3c in do_system () from /lib64/libc.so.6
#2  0x0000100002b40424 in system () from /lib64/libc.so.6
#3  0x0000100002ac5348 in system () from /lib64/libpthread.so.0
#4  0x0000100001cc9714 in TUnixSystem::StackTrace() () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-06-2300/external/slc7_ppc64le_gcc820/lib/libCore.so
#5  0x0000100001cccd18 in TUnixSystem::DispatchSignals(ESignals) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-06-2300/external/slc7_ppc64le_gcc820/lib/libCore.so
#6  0x0000100001ccce04 in SigHandler(ESignals) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-06-2300/external/slc7_ppc64le_gcc820/lib/libCore.so
#7  0x0000100001cc4444 in sighandler(int) () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-06-2300/external/slc7_ppc64le_gcc820/lib/libCore.so
#8  <signal handler called>
#9  0x0000100002989fd0 in fesetenv

GLIBC_2.17 () from /lib64/libm.so.6
#10 0x00001000024ab6a0 in tbb::internal::cpu_ctl_env::set_env (this=<optimized out>) at ../../src/tbb/scheduler_common.h:446
#11 tbb::internal::context_guard_helper<false>::set_ctx (ctx=<optimized out>, this=0x3fffc4e842e0) at ../../src/tbb/scheduler.h:993
#12 tbb::internal::custom_scheduler<tbb::internal::IntelSchedulerTraits>::process_bypass_loop (this=0x1000037fb200, context_guard=..., t=0x100140737340, isolation=0) at ../../src/tbb/custom_scheduler.h:462
#13 0x00001000024abaa4 in tbb::internal::custom_scheduler<tbb::internal::IntelSchedulerTraits>::local_wait_for_all (this=0x1000037fb200, parent=..., child=<optimized out>) at ../../include/tbb/task.h:1003
#14 0x00001000024ac4b0 in tbb::internal::custom_scheduler<tbb::internal::IntelSchedulerTraits>::wait_for_all (this=<optimized out>, parent=..., child=0x0) at ../../src/tbb/governor.h:123
#15 0x00001000002871b4 in edm::eventsetup::EventSetupRecordIOVQueue::~EventSetupRecordIOVQueue() () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-06-2300/lib/slc7_ppc64le_gcc820/libFWCoreFramework.so
#16 0x000010000021fa14 in edm::EventProcessor::~EventProcessor() () from /cvmfs/cms-ib.cern.ch/nweek-02653/slc7_ppc64le_gcc820/cms/cmssw/CMSSW_11_2_X_2020-11-06-2300/lib/slc7_ppc64le_gcc820/libFWCoreFramework.so
#17 0x000000001000f298 in (anonymous namespace)::EventProcessorWithSentry::~EventProcessorWithSentry() ()
#18 0x000000001000ea74 in main ()
=========================================================== 
```

reproducing it requires access to one of the ppc machines,
and then:

```
source /cvmfs/cms-ib.cern.ch/week1/cmsset_default.sh
scram p CMSSW_11_2_X_2020-11-05-2300 # last IB without the failure
cd CMSSW_11_2_X_2020-11-05-2300/src
git-cms-addpkg PhysicsTools/PatAlgos;
scram b -j 64 ; scram b runtests_runtestPhysicsToolsPatAlgos # runs
git-cms-merge-topic 32049
scram b -j 64 ; scram b runtests_runtestPhysicsToolsPatAlgos # fails
```

**Comment by Dr15Jones** [comment]
@mrodozov How did you determine it was this pull request? The code removed was not being used anywhere in CMSSW (except a handful of tests in FWCore/MessageService).

**Comment by Dr15Jones** [comment]
@mrodozov Your test would also have changed memory layout of the job which could have triggered a problem that has existed all along.

**Comment by Dr15Jones** [comment]
All the failures (at least in the RelVals) are floating point errors. Very weird.

**Comment by mrodozov** [comment]
well I took the last IB without that and put the 6 PRs between it and when it started to fail to check after which it started to fail

**Pull request merged**